File: JSVaporizer.NET.8/wwwroot/jsvwasm/for_dotnet/window.js
"use strict";

let jsvExports;

export function getWindow(exports) {
    jsvExports = exports;
    return {
        alert: (s) => alert(s),
        location: {
            href: () => window.location.href
        },
        console: {
            log: (str) => console.log(str),
            dir: (obj) => console.dir(obj)
        }
    };
}

File: JSVaporizer.NET.8/JSVaporizer/BrowserJSFacade/Window.cs
using System.Runtime.Versioning;

namespace JSVaporizer;

[SupportedOSPlatform("browser")]
internal static partial class JSVapor
{
    public static class Window
    {
        public static string Alert(string text)
        {
            return WasmWindow.Alert(text);
        }

        public static class Location
        {
            public static string Href()
            {
                return WasmWindow.Location.Href();
            }
        }
    }

    public static class Console
    {
        public static void Log(string str)
        {
            WasmWindow.Console.Log(str);
        }

        public static void Dir(object obj)
        {
            WasmWindow.Console.Dir(obj);
        }
    }
}


